core/string/uminus_spec.rb

```diff
@@ -17,5 +17,18 @@
       output.frozen?.should == true
       output.should == 'foo'
     end
+
+    ruby_version_is "2.5" do
+      it "returns the same object for equal unfrozen strings" do
+        origin = -"this string is frozen"
+        dynamic = -%w(this string is frozen).join(' ')
+
+        origin.should equal dynamic
+      end
+
+      it "returns the same object when it's called on the same String literal" do
+        (-"unfrozen string").should equal(-"unfrozen string")
+      end
```

Comment:
Maybe an extra spec to make sure it returns a different object for different strings? :D

Reply:
@eregon Done.
The features aren't described clearly in the tickets (IMHO) and I haven't found any related tests in the `ruby/ruby` git repository. So without diving into source code I cannot say these specs are comprehensive